clients/drcachesim/tracer/output.cpp

```diff
@@ -1022,9 +1033,18 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
             }
             if (hit_window_end) {
                 if (op_offline.get_value() && op_split_windows.get_value()) {
-                    size_t add =
-                        instru->append_thread_exit(buf_ptr, dr_get_thread_id(drcontext));
-                    buf_ptr += add;
+                    if (op_L0_filter_until_instrs.get_value() &&
+                        tracing_mode.load(std::memory_order_acquire) ==
+                            BBDUP_MODE_L0_FILTER) {
+                        NOTIFY(0, "Adding endpoint marker\n");
+                        size_t add = instru->append_marker(
+                            buf_ptr, TRACE_MARKER_TYPE_FILTER_ENDPOINT, 0);
```

Comment:
For -split_windows we have a separate file per window: so the filter will be its own trace, with its own filetype showing that it's filtered.  So it seems we do not need this endpoint marker: not seeing what value it adds?  I remember @abhinav92003 adding it but not remembering why.

Reply:
Sorry for the confusion but I don't have separate file for filter and full traces. So the endpoint marker is inserted between the filtered trace and the full trace. I have it in the raw trace as well as the final trace. Since @abhinav92003 is filtering the converted trace, he did not need it in the raw trace, but he also has the two types of entries in the same file.

Reply:
> Could you clarify: do you mean that if -split_windows is set you ignore it and do not create a separate trace for each window?

> Re: single file with multiple windows: agreed, the endpoint marker does make sense there. Probably we should just leave it in the separate files too for simplicity and consistency -- I assume you do get separate files for -split_windows right?

No I don't ignore it. We get a separate file for each window, and this file contains both filtered and full trace entries separated by an endpoint marker. Does that make sense?

Reply:
Yes, all cleared up: I think I misread your "I don't have separate file for filter and full" as not having separate files at all.